frontend/src/pages/dashboard/profile/Profile.tsx

```diff
@@ -1,189 +1,83 @@
 import React, { useEffect } from "react";
 import "./Profile.scss";
-import completing from "../../../illustrations/completing.svg";
 import { useAlert } from "react-alert";
 import { axiosError } from "../../../def/axios-error";
+import Button from "../../../components/button/Button";
+import { ProfileDef } from "../../../def/ProfileDef";
+import Section from "../../../components/section/Section";
+import Avatar from "../../../components/avatar/Avatar";
+import Skeleton from "react-loading-skeleton";
 
 /**
  * Renders the profile page, currently static
  */
 export default function Profile(props: {}) {
 	const alert = useAlert();
 	
+	const [profile, setProfile] = React.useState<ProfileDef | undefined>(undefined);
+	const [updateUsernameInputValue, setUpdateUsernameInputValue] = React.useState("");
+	
 	useEffect(() => {
-		global.axios.get("profile", { withCredentials: true })
-			  .then(res => console.log(res))
+		global.axios.get<any>("profile", { withCredentials: true })
+			  .then(res => {
+				  setProfile({
+					  id: res.data.userId ?? "",
+					  name: res.data.username ?? "",
+					  type: res.data.accountState ?? "",
+					  registrationDate: res.data.registrationDate ?? ""
+				  });
+				  setUpdateUsernameInputValue(res.data.username ?? "");
+			  })
 			  .catch(err => axiosError(err, alert));
 	}, [alert]);
 	
 	return <>
-		<div className={ "container profile-main" }>
-			<div style={ { display: "flex", gap: "var(--spacing)" } }>
-				<div style={ { flex: 1 } }>
-					<h1>
-						<i className={ "fi fi-rr-user" }/>
-						Profile
-					</h1>
-					<p>View and edit your profile.</p>
-					
-					<hr/>
-					<h3>Profile Picture</h3>
-					<div style={ { display: "flex", gap: "var(--spacing)" } }>
-						<img src={ "https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50" }
-							 alt={ "Avatar" }
+		<Section className={ "transparent" } style={ { flexDirection: "column" } }>
+			{ profile
+				? <Avatar userId={ profile.id } style={ { height: 200, width: 200 } }/>
+				: <Skeleton width={ 200 } height={ 200 } style={ { borderRadius: "50%" } }/>
+			}
+			<div style={ { display: "flex", alignItems: "center", gap: "var(--spacing)" } }>
+				{ profile
+					? <input id={ "test" }
+							 minLength={ 4 } maxLength={ 20 }
 							 style={ {
-								 height: "100px",
-								 width: "100px",
-								 borderRadius: "50%",
-								 objectFit: "cover"
+								 background: "transparent",
+								 border: "none",
+								 borderRadius: "var(--border-radius)",
+								 fontSize: "2em",
+								 padding: "calc(var(--spacing) / 2) var(--spacing)",
+								 fontWeight: 700,
+								 width: 400
+							 } }
+							 placeholder={ "Username" }
+							 value={ updateUsernameInputValue }
+							 onChange={ (e) => setUpdateUsernameInputValue(e.target.value.trim()) }
+							 onFocus={ (e) => {
+								 window.setTimeout(() => document.execCommand('selectAll', false, undefined), 1);
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand

Reply:
No critical deprecation. Also this has been in the code from the start. Eventually we might consider switching to a rich text editor library.